urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/util/ByteArrayComparator.java

```diff
@@ -0,0 +1,93 @@
+package ikrs.util;
+
+import java.util.Comparator;
+
+/**
+ * A very simple byte array comparator.
+ *
+ * @author Ikaros Kappler
+ * @date 2013-04-10
+ * @version 1.0.0
+ **/
+
+
+public class ByteArrayComparator 
+    implements Comparator<byte[]> {
+
+    //--- BEGIN ----------------- Comparator ------------------------
+    /**
+     * Compared the passed arrays.
+     *
+     * See compareArrays( o1, o2 ).
+     *
+     * @see compareArrays( byte[], byte[] )
+     **/
+    public int compare( byte[] o1, byte[] o2 ) {
+	return ByteArrayComparator.compareArrays( o1, o2 );
+    }
+    
+    public boolean equals( Object obj ) {
+	
+	try {
+	    return (obj instanceof ByteArrayComparator);
+	} catch( ClassCastException e ) {
+	    return false;
+	}
+	
+    }
+    //--- END ------------------- Comparator ------------------------
+
+    /**
+     * Compares the passed byte arrays the way the data would
+     * be compared in a string: byte for byte instead of char by char,
+     * beginning at index 0.
+     *
+     * If one array is prefix of the other, the longest is considered
+     * to be 'larger'.
+     *
+     * null is considered to be 'smaller' than non-null.
+     *
+     * null equals null.
+     *
+     * If both arrays have the same length and all bytes are equal
+     * both arrays are considered equal.
+     * 
+     *
+     * @param o1 The first byte array.
+     * @param o2 The second byte array.
+     **/
+    public static int compareArrays( byte[] o1, byte[] o2 ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 73-81, lines 84-89 into separate methods.